### ISSUE TITLE
Input to output leakage during training

### ISSUE THREAD

**Issue by Zeyad-Abdelreheem** [Chat-3D/Chat-3D-v2#49]
I observed that during the training phase, the answer is embedded and added to the input embeddings fed to the LLM ([here](https://github.com/ZzZZCHS/Chat-Scene/blob/5da1d87003664edbf3292cc888d3d81964c22bfd/models/chat3d.py#L412)). Is this intentional? If so, could it lead to data leakage or overfitting, potentially degrading performance during inference when the answer is not provided?

**Comment by ZzZZCHS** [comment]
Hi, thanks for your interest.

It's standard practice to feed answer tokens into the LLM for [calculating loss](https://github.com/ZzZZCHS/Chat-Scene/blob/dev/models/modeling_llama.py#L1226-L1237) on each answer token. The causal attention mask ensures that LLM can only see previous tokens when predicting the next token. Therefore, I don't believe there is any risk of data leakage.
